### ISSUE TITLE
`phi-is-not-first` only reports the last misplaced `φ` in the document, ignoring multiple formations per file

### ISSUE THREAD

**Issue by gemshrine** [objectionary/lints#1359]
`phi-is-not-first` takes the last misplaced `φ` in the **whole document**, so when two different formations each have one, only the second is reported.

`src/main/resources/org/eolang/lints/names/phi-is-not-first.xsl:13`:

```xml
<xsl:apply-templates select="(//o[@name='φ'][preceding-sibling::o[not(@base='∅') and not(@base='ξ' and @name='xi🌵')]])[last()]" mode="unordered"/>
```

`//o[...]` gathers every misplaced `φ` in the file; the parentheses make `[last()]` apply to that whole sequence, not to each formation's own. Whatever the file contains, exactly one defect comes out.

Reporting one per formation is deliberate — `packs/single/phi-is-not-first/catches-only-last-phi.yaml` pins it, with two misplaced `φ` inside one formation and `count(defect)=1` expected. Every pack in that directory has a single formation, so the behaviour that is pinned is the within-formation one.

Across formations the same expression does something nobody asked for:

```
# No comments.
[] > foo
  y > bar
  x > @

# No comments.
[] > bar
  y > baz
  x > @
```

`foo`'s misplaced `φ` is dropped and only `bar`'s is reported. A file with ten such formations yields one defect, and each fix reveals the next one on the following run.

Scoping `last()` to the formation rather than to the document keeps the pinned behaviour and restores the rest:

```xml
<xsl:for-each select="//o[o[@name='φ'][preceding-sibling::o[not(@base='∅') and not(@base='ξ' and @name='xi🌵')]]]">
  <xsl:apply-templates select="(o[@name='φ'][preceding-sibling::o[not(@base='∅') and not(@base='ξ' and @name='xi🌵')]])[last()]" mode="unordered"/>
</xsl:for-each>
```

so `[last()]` picks the last offender within each parent, and every parent gets its own defect.

A pack with two formations, each carrying a misplaced `φ`, expecting `count(defect)=2`, would keep the two cases apart — the existing packs cannot tell the difference, since they never have a second formation to lose.

**Comment by 0crat** [comment]
@gemshrine Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+3650](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
